.github/workflows/sonar.yml

```diff
@@ -4,39 +4,45 @@ on:
   push:
     branches:
       - main
-      - user/markackert/sonar
   pull_request:
     branches:
-      - "**"
+      - '**'
+
+permissions: read-all
 
 jobs:
   sonar:
     runs-on: ubuntu-latest
 
     steps:
-    - name: Checkout
-      uses: actions/checkout@v2
-      with:
-        fetch-depth: 0 
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Use Node.js LTS
+        uses: actions/setup-node@v2
```

Comment:
if possible, we should consider updating these actions to v3 since some of them may already be deprecated